x-pack/plugins/apm/public/components/app/service_details/service_icons/cloud_details.tsx

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiBadge, EuiDescriptionList } from '@elastic/eui';
+import { EuiDescriptionListProps } from '@elastic/eui/src/components/description_list/description_list';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { APIReturnType } from '../../../../services/rest/createCallApmApi';
+
+type ServiceDetailsReturnType = APIReturnType<'GET /api/apm/services/{serviceName}/metadata/details'>;
+
+interface Props {
+  cloud: ServiceDetailsReturnType['cloud'];
+}
+
+export function CloudDetails({ cloud }: Props) {
+  if (!cloud) {
+    return null;
+  }
+
+  const listItems: EuiDescriptionListProps['listItems'] = [];
+  if (cloud.provider) {
+    listItems.push({
+      title: i18n.translate('xpack.apm.serviceNameHeader.cloud.provider', {
+        defaultMessage: 'Cloud provider',
+      }),
+      description: cloud.provider,
+    });
+  }
+
+  if (!!cloud.availabilityZones?.length) {
+    listItems.push({
+      title: i18n.translate('xpack.apm.serviceNameHeader.cloud.zone', {
+        defaultMessage:
+          '{zones, plural, =0 {Availability zone} one {Availability zone} other {Availability zones}} ',
+        values: { zones: cloud.availabilityZones.length },
+      }),
+      description: (
+        <ul>
+          {cloud.availabilityZones.map((zone, index) => (
+            <li key={index}>
+              <EuiBadge color="hollow">{zone}</EuiBadge>
+            </li>
+          ))}
+        </ul>
+      ),
+    });
+  }
+
+  if (cloud.machineTypes) {
+    listItems.push({
+      title: i18n.translate('xpack.apm.serviceNameHeader.cloud.machine', {
+        defaultMessage:
+          '{machineTypes, plural, =0{Machine type} one {Machine type} other {Machine types}} ',
```

Comment:
Same question here as with AZs.

Reply:
Same thing here, if there are no machine types this section won't show.